tests/integration/test_utils/data_source_config/base.py

```diff
@@ -142,8 +147,14 @@ def _random_resource_name() -> str:
 
     @cached_property
     def context(self) -> AbstractDataContext:
+        if self._context is not None:
```

Comment:
Nit: Now that we're setting this in the constructor, the cached_property probably doesn't get us anything. I'd just remove this and do `self.context = context or gx.get_context()` in the constructor (or make that field non-nullable like mentioned above.)

Reply:
i buy it: [1665fcf](https://github.com/great-expectations/great_expectations/pull/11138/commits/1665fcf2362454431fdce9e2b7dacc2729f8c530)